assembly/__tests__/runtime/avl-tree.spec.ts

```diff
@@ -0,0 +1,544 @@
+import { AVLTree } from "../../runtime";
+
+let tree: AVLTree<u32, u32>;
+
+// Return height of the tree - number of nodes on the longest path starting from the root node.
+function height<K, V>(tree: AVLTree<K, V>): u32 {
+    throw new Error("TODO implement height()");
+}
+
+function random(n: i32): u32[] {
+    const a = new Array<u32>(n);
+    // FIXME need rng
+    return a.map<u32>((_): u32 => 5);
+}
+
+function range<T>(start: T, end: T): T[] {
+    const a = new Array<T>();
+    let x = start;
+    while (x < end) {
+        a.push(x);
+    }
+    return a;
+}
+
+function maxTreeHeight(n: f64): u32 {
+    // From near-sdk-rs TreeMap:
+    // h <= C * log2(n + D) + B
+    // where:
+    // C =~ 1.440, D =~ 1.065, B =~ 0.328
+    // (source: https://en.wikipedia.org/wiki/AVL_tree)
+    const B: f64 = -0.328;
+    const C: f64 = 1.440;
+    const D: f64 = 1.065;
+
+    const h = C * Math.log2(n + D) + B;
```

Comment:
This uses floating point numbers which aren't allowed, so you should use `math.binaryLog` instead, which only uses integers.

Reply:
Ah never mind this is valid in a test file, but is true the actual file.